### ISSUE TITLE
python3

### ISSUE THREAD

**Issue by rabinovic** [technic/enigma2_example#3]
Hi, can you give me aan hit how to switch to python3? 
Do you have any intention to deploy a Dockerfile with enigma2 based on python3?

Thanks :)

**Comment by technic** [comment]
Hi, I don't use python3 in enigma2 yet.
Firstly, this enigma2 fork is openpli based, do they have python3 already? Then one need to rebase few patches from this fork onto the upstream enigma2. Mainly for rendering gstreamer player to the SDL2 window (you can skip this if you don't need it). There were few other small patches, which I can't remember now =)
Then in general building and docker configuration should straight forward, just by replacing python with python3 everywhere.

**Comment by technic** [comment]
Btw, this repository is a bit out dated. There is a new remote devcontainer feature in vscode, which allows to achieve same experience easier. But for my own purposes I stick with this option, since it is already configured on my PC.

**Comment by rabinovic** [comment]
Hi again,
i forged your enigma2 repo and switched to python 3. It compiles well but unfortunately when i run enigma2 it it stops without any error. I would like to ask you if you could take a look at the following docker file and my fork and see why enigma2 does not run constantly:
Dockerfile
-------------

FROM ubuntu:20.04

ARG DEBIAN_FRONTEND=noninteractive

RUN apt-get -q update && apt-get install -y g++-10 gcc-10 linux-libc-dev git build-essential \
    automake libtool python3.9-dev zlib1g-dev gettext swig libgstreamer1.0-dev \
    libgstreamer-plugins-base1.0-dev libfreetype6-dev libsigc++-2.0-dev libfribidi-dev \
    libssl-dev libavahi-client-dev libjpeg-turbo8-dev libgif-dev python3-pip libsdl2-dev libxml2-dev \
    x11vnc xvfb xdotool apache2 \
    libarchive-dev libcurl4-openssl-dev libgpgme11-dev libtool libtool-bin

RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-10 100 \
    && update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-10 100

RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.9 1 \
    && update-alternatives --install /usr/bin/pip pip /usr/bin/pip3 1

RUN pip3 install python-wifi

WORKDIR /work

ARG OPKG_VER="0.3.5"
RUN curl "http://git.yoctoproject.org/cgit/cgit.cgi/opkg/snapshot/opkg-$OPKG_VER.tar.gz" -o opkg.tar.gz
RUN tar -xzf opkg.tar.gz \
 && cd "opkg-$OPKG_VER" \
 && ./autogen.sh \
 && ./configure --enable-curl --enable-ssl-curl --enable-gpg \
 && make \
 && make install

RUN git clone --depth 1 git://git.opendreambox.org/git/obi/libdvbsi++.git
RUN cd libdvbsi++ \
 && ./autogen.sh \
 && ./configure \
 && make \
 && make install

RUN git clone --depth 1 git://github.com/OpenPLi/tuxtxt.git
RUN cd tuxtxt/libtuxtxt \
 && autoreconf -i \
 && CPP="gcc -E -P" ./configure --with-boxtype=generic DVB_API_VERSION=5 --prefix=/usr \
 && make \
 && make install

 RUN cd tuxtxt/tuxtxt \
 && autoreconf -i \
 && CPP="gcc -E -P" ./configure --with-boxtype=generic DVB_API_VERSION=5 --prefix=/usr \
 && make \
 && make install

RUN git clone --depth 1 https://github.com/rabinovic/enigma2.git
RUN cd enigma2 \
 && autoreconf -i \
&& ./configure --with-libsdl --with-gstversion=1.0 --prefix=/usr --sysconfdir=/etc \
 && make -j4 \
 && make install

COPY enigma2-settings /etc/enigma2/settings
RUN ldconfig

RUN pip3 install ptvsd

EXPOSE 5678
COPY mytest-ptvsd.py /opt/mytest-ptvsd.py

COPY entrypoint.sh /opt
RUN chmod 755 /opt/entrypoint.sh
ENV DISPLAY=:99
EXPOSE 5900 80
ENTRYPOINT ["/opt/entrypoint.sh"]

CMD [ "x11vnc", "-forever"]

**Comment by technic** [comment]
Well, that's a lot of changes that you've made. Why don't you use openpli python3 branch?

**Comment by rabinovic** [comment]
I have also tried that branch, but with the same result.

**Comment by technic** [comment]
> I have also tried that branch, but with the same result.

Can you push that to the repo, so I can have a look at it

**Comment by rabinovic** [comment]
The repo is now identical with the openpli python3 branch except the changes concerning sdl2.

**Comment by technic** [comment]
Damn, there are missing `\` in the Dockerfile you pasted above

**Comment by technic** [comment]
btw, do we really need gcc10 and python3.9, are these the versions used in enigma2 openpli toolchain?

**Comment by technic** [comment]
i can not compile this https://github.com/rabinovic/enigma2.git

**Comment by rabinovic** [comment]
yes, it won't compile unless i revert the last commited changes. 
I think that version https://github.com/rabinovic/enigma2/tree/25387f850fd9681df546ba49cd9df8a661811695 
should compile.

**Comment by technic** [comment]
The `Tools.RedirectOutput` doesn't work with python3 for some reason. Comment this import in `StartEnigma.py` and you will see the python traceback causing the error.

Also launch like this: `ENIGMA_DEBUG_LVL=5 enigma2`

I think the python3 support in openPLi is still not ready to use.
Upd: ...so maybe it's better to try out openatv.

**Comment by technic** [comment]
The OpenATV based Python 3 version is available in [new branch](https://github.com/technic/e2xvfb/tree/atv-py3)

**Comment by rabinovic** [comment]
Great job. Thanks
